messages.go

```diff
@@ -0,0 +1,140 @@
+package dht
+
+import (
```

Comment:
@aschmahmann We should call this `dht_rpc_client.go` if you like the name ?

Reply:
🤷 I was trying to avoid the rpc terminology, but if you think that is more intuitive then it works for me